### PR TITLE
codec/datum.rs:prepare for eval_json

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
    This PR do some preparation in datum.rs for evaluate json functions.
@andelf @lishihai9017 @hhkbp2 @hicqu  PTAL

### CODE REVIEW

**Comment by hhkbp2** [comment]
LGTM

**Comment by hicqu** [comment]
LGTM.

File: src/util/codec/datum.rs [hunk 1]
```diff
@@ -1341,4 +1403,55 @@ mod test {
             assert_eq!(res_y, exp_y);
         }
     }
+
+    #[test]
+    fn test_cast_as_json() {
+        let tests = vec![
+            (Datum::I64(1), "1.0"),
+            (Datum::F64(3.3),"3.3"),
```
Comment (BusyJay) [inline src/util/codec/datum.rs]: space.

File: src/util/codec/datum.rs [hunk 2]
```diff
@@ -1341,4 +1403,55 @@ mod test {
...
+        for d in illegal_cases {
+            assert!(d.cast_as_json().is_err());
+        }
+    }
+
+    #[test]
+    fn test_datum_into_json() {
+        let tests = vec![
+            (Datum::I64(1), "1.0"),
+            (Datum::F64(3.3),"3.3"),
```
Comment (BusyJay) [inline src/util/codec/datum.rs]: Space.

File: src/util/codec/datum.rs [hunk 3]
```diff
@@ -299,23 +299,32 @@ impl Datum {
         Ok(b)
     }
 
-    /// into_string convert self into a string.
-    /// source function name is `ToString`.
-    pub fn into_string(self) -> Result<String> {
-        let s = match self {
+    pub fn as_string(&self) -> Result<String> {
```
Comment (BusyJay) [inline src/util/codec/datum.rs]: `to_string` is more appropriate.

File: src/util/codec/datum.rs [hunk 4]
```diff
@@ -399,6 +408,59 @@ impl Datum {
...
+            Datum::Null => Ok(Json::None),
+            Datum::Bytes(bs) => {
+                let s = try!(String::from_utf8(bs));
+                Ok(Json::String(s))
+            }
+            _ => self.cast_as_json(),
+        }
+    }
+
+    pub fn as_json_path_expr(&self) -> Result<PathExpression> {
```
Comment (BusyJay) [inline src/util/codec/datum.rs]: `to_json_path_expr` is more appropriate.

**Pull request merged**